client/app/src/main/java/com/microsoft/research/karya/ui/dashboard/DashboardViewModel.kt

```diff
@@ -41,22 +61,96 @@ constructor(
     MutableStateFlow(DashboardUiState.Success(emptyList()))
   val dashboardUiState = _dashboardUiState.asStateFlow()
 
-  fun fetchNewTasks() {
+  fun syncWithServer() {
     viewModelScope.launch {
-      val idToken = authManager.fetchLoggedInWorkerIdToken()
+      submitCompletedTasks()
+      fetchNewTasks()
+      cleanupKaryaFiles()
+    }
+  }
+
+  suspend fun fetchNewTasks() {
+    val idToken = authManager.fetchLoggedInWorkerIdToken()
+
+    // Get Assignment DB updates
+    assignmentRepository.getAssignments(idToken, "new", "").catch { throw it }.collect()
```

Comment:
Don't throw error directly, send state update like this `_dashboardUiState.value = DashboardUiState.Error(it)`